### PR TITLE
vmware esxi filter plugin, conf and test cases

### PR DESCRIPTION
**Pull request by arjun-urs**
Hello Team,

Code includes:
1. configuration file to listen, filter and output.
2. filter plugin to tokenize and add common fields.
3. Test cases.

Please review.

@Microsoft/omsagent-devs @anugup @keikhara @amitsara

### CODE REVIEW

File: installer/conf/omsagent.d/vmware_esxi.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<source>
+  type syslog
+  format /^(?<EventTime>\S+)\s{0,4}(?<HostName>\S+)\s{0,4}(?<ProcessName>\S+)\:\s{0,4}(?<SyslogMessage>.*)$/
+  port 1514
```
Comment (robbiezhang) [inline installer/conf/omsagent.d/vmware_esxi.conf]: why do you use this port? how can the user configure the rsyslog to forward the VMWare syslog to this port?

**Comment by keikhara** [comment]
https://github.com/Microsoft/OMS-Agent-for-Linux/blob/keikodoc/docs/VMware-Instruction.md 

customer will set it up on ESXi Host.

File: installer/conf/omsagent.d/vmware_esxi.conf (hunk 1 above)
Comment (keikhara) [inline installer/conf/omsagent.d/vmware_esxi.conf]: https://github.com/Microsoft/OMS-Agent-for-Linux/blob/keikodoc/docs/VMware-Instruction.md

customer will set it up on ESXi Host.

**Comment by robbiezhang** [comment]
please squash the 2 commits into 1. you can use "git rebase -i".
http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

File: source/code/plugins/filter_vmware_logs.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      # Co-relation fields
+      record['Computer'] = record['HostName']
+      record['ResourceName'] = 'VMware'
+      record['ResourceLocation'] = 'VMware'
+      record['ResourceType'] = 'Hypervisor'
+      record['ResourceId'] = record['HostName']
+
+      begin
+        # Regex for Device. Example string: naa.60a9800041764b6c463f43786855a3t2
+        record['Device'] = record['SyslogMessage'].to_s.match(/naa\.[a-z0-9]{32}/).to_s
```
Comment (robbiezhang) [inline source/code/plugins/filter_vmware_logs.rb]: I would recommend you extract the logic into a parser plugin that the syslog plugin can apply.
http://docs.fluentd.org/articles/parser-plugin-overview#

You can use the built-in record_transform plugin for adding, removing or modifying fields.
http://docs.fluentd.org/articles/filter_record_transformer

**Comment by arjun-urs** [comment]
Code updated according to the comments. Please review.

**Comment by robbiezhang** [comment]
Signoff

**Pull request merged**